# Bioinformatics Tutorials that use CyVerse

## VICE

| Tutorial	| Date | Notes |
|-----------|------|-------|
| [RNASeq using VICE](https://rnaseq-webinar-2019.readthedocs.io/en/latest/){target=_blank} | Nov. 8, 2019	| Perform RNAseq differential expression analysis using Read Mapping and Transcript Assembly (RMTA) and Rstudio-DESEq2 apps |

## Discovery Environment

| Tutorial | Date |	Notes |
|----------|------|-------|
| [Assemble a Genome Using SOAPdenovo](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=14588299){target=_blank}	| Dec 19, 2019 | Commonly used procedure for de novo whole genome assembly of Illumina reads using the DE: Assemble reads, Assess assembly |
| [BLAST a Transcriptome](https://wiki.cyverse.org/wiki/display/TUT/Tutorial%3A+BLAST+a+Transcriptome){target=_blank} | May 11, 2016 |	Reduce number of transcripts and level of redundancy in an assembled transcriptome, and identify coding sequences that can be submitted to BLASTP searches. |
| [Characterizing Differential Expression With RNA-Seq (Without Reference Genome)](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=14585157){target=_blank}	| Jul 21, 2015 | Identify changes in gene expression levels between at least two sequenced transcriptome samples (18 separate tutorials) |
| [Cluster Orthologs and Paralogs and Assemble Custom Gene Sets](https://cyverse.atlassian.net/wiki/spaces/TUT/pages/258736290/Cluster+Orthologs+and+Paralogs+and+Assemble+Custom+Gene+Sets+Workflow+Tutorial){target=_blank} | Dec 11, 2019 | Input entire protein-encoding gene or transcript repertoires from genomes of interest, and cluster homologs (orthologs and paralogs), then query clusters to assemble gene sets based on presence/absence and copy number. |
| [Evaluate and Pre-Process Sequencing Reads](http://www.cyverse.org/learning-center/discovery-environment/de-004-genome-wide-association-study-gwas-using-genotyping){target=_blank} | Jan 05, 2018 |	Clean and filter Illumina reads using DE apps. |
| [Evaluate High-throughput Sequencing Reads with FastQC](https://cyverse-fastqc-quickstart.readthedocs-hosted.com/en/latest/){target=_blank} | |	FastQC is a popular tool for evaluating the quality of high-throughput sequencing reads such as from Illumina and PacBio. |
| [Filter, Trim, and Process High-throughput Sequencing Reads with Trimmomatic](https://cyverse-trimmomatic-quickstart.readthedocs-hosted.com/en/latest/){target=_blank} | Sep 15, 2016 |	Trimmomatic is a popular application for filtering and trimming high- throughput sequencing reads. Several functions can remove populations of low quality reads, remove sequencing adaptors, and trim low-quality regions of individual reads. |
| [Genome Annotation with MAKER](https://cyverse-sciapps-guide.readthedocs-hosted.com/en/latest/annotation.html){target=_blank} | Oct 09, 2019 | This tutorial is a step-by-step guide for using SciApps to perform MAKER based annotation |
| [Genome-wide Association Study (GWAS) Using a Genotyping-by-sequencing Approach](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=11447702){target=_blank} | |	Learn to identify genetic variants that are associated with a trait. | 
| [HTSeqQC](https://learning.cyverse.org/projects/cyverse-htseqqc-cyverse-tutorial/en/latest/){target=_blank} | Aug 01, 2017 |	An automated quality control analysis tool for a single and paired-end high-throughput sequencing data (HTS) generated from Illumina sequencing platforms | 
| [Import data from NCBI SRA using the Discovery Environment](http://learning.cyverse.org/projects/cyverse-importing-sradata-quickstart/){target=_blank}	| | The NCBI Sequence Read Archive (SRA) is a repository for high-throughput sequencing reads. These are valuable data for novel analysis and reuse. You can directly import data from SRA into your Data Store using a Discovery Environment app. |
| [Kallisto-0.42.3-INDEX-QUANT-PE](https://wiki.cyverse.org/wiki/display/TUT/Kallisto-0.42.3-INDEX-QUANT-PE+in+the+Discovery+Environment){target=_blank} | Oct 25, 2019 |	Kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment. |
| [mini SOAPdenovo](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=22676743){target=_blank} | Jan 04, 2016 |	Gain familiarity with a commonly used procedure for de novo whole genome assembly of Illumina reads using the DE. |
| [QIIME-1.9.1 for the DE](https://wiki.cyverse.org/wiki/display/TUT/QIIME-1.9.1+in+Discovery+Environment){target=_blank} | Apr 12, 2016 |	QIIME is an open-source bioinformatics pipeline for performing microbiome analysis from raw DNA sequencing data. |
| [RNA-Seq with Kallisto and Sleuth](https://cyverse-kallisto-tutorial.readthedocs-hosted.com/en/latest/){target=_blank} | Nov 04, 2019 |	Kallisto is a quick, highly-efficient software for quantifying transcript abundances in an RNA-Seq experiment. Sleuth is designed to analyze and visualize the Kallisto results in R. |
| [Submit High-throughput Sequencing Reads to NCBI Sequence Read Archive (SRA)](https://learning.cyverse.org/projects/sra_submission_quickstart/en/latest/){target=_blank} | Dec 04, 2017|	The SRA is a canonical repository for sequencing data generated by high-throughput instruments. The CyVerse submission pipeline allows you to directly submit your data into an SRA-linked BioProject. |
| [Taxonomic Name Resolution Service (TNRS)](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=11447869){target=_blank} | DEc 05, 2017 |	Become familiar with TNRS to identify, correct, and update scientific names of plants. |
| [Discover Variants Using SAM Tools](https://learning.cyverse.org/en/latest/tutorials.html#system-message-1){target=_blank} | Oct 11, 2016 |	Detect and call variants from sequence reads using Bowtie and SAM Tools. |

## SciApps

| Tutorial | Date |	Notes |
|----------|------|-------|
| [Association analysis with mixed models](https://cyverse-sciapps-guide.readthedocs-hosted.com/en/latest/association.html){target=_blank}	| | A genome-wide association study (or GWAS) workflow using TASSEL, EMMAX, and MLMM for mixed model analysis.

## Atmosphere

| Tutorial | Date |	Notes |
|----------|------|-------|
| [Basic Stacks](https://cyverse.atlassian.net/wiki/spaces/TUT/pages/258736150/Basic+Stacks+Atmosphere+Images+Tutorial){target=_blank} | Nov 06, 2017| Use next generation sequence data produced from Reduced Representation Libraries (RRL) such as Restriction site associated (RAD) tags. |
| [BATools 0.0.1](https://wiki.cyverse.org/wiki/pages/editpage.action?pageId=11443702){target=_blank} | Apr 10, 2016 |	Introduce new users to BATools and the BATools Wrapper Script. |
| [Evolinc](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=22677526){target=_blank} | May 03, 2016|	Evolinc is a two-part pipeline to identify lincRNAs from an assembled transcriptome file (.gtf output from cufflinks) and then determine the extent to which those lincRNAs are conserved in the genome and transcriptome of other species. |
| [FaST-LMM.Py v2.02](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=12884250){target=_blank} | Apr 19, 2016 |	Introduce new users to the FaST-LMM software for GWAS analysis. |
| [fastStructure](https://wiki.cyverse.org/wiki/display/TUT/fastStructure+using+Atmosphere){target=_blank} | Oct 01, 2017|	fastStructure is a fast algorithm for inferring population structure from large SNP genotype data. It is based on a variational Bayesian framework for posterior inference and is written in Python2.x. |
| [Installing R packages on Atmosphere](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=11445211){target=_blank} | Jun 23, 2016|	Install R packages on Atmosphere: Launch instance, transfer files to instance, install R package, request imaging. |
| [KOBAS 2.0-09052014](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=12223187){target=_blank} | Apr 19, 2016|	Learn how to annotate and identify using KOBAS 2.0. |
| [QIIME-1.9.1 for Atmosphere](https://wiki.cyverse.org/wiki/display/TUT/QIIME-1.9.1+using+Atmosphere){target=_blank} | Jun 12, 2017 |	QIIME is an open-source bioinformatics pipeline for performing microbiome analysis from raw DNA |sequencing data. QIIME is designed to take users from raw sequencing data generated on the Illumina or other platforms through publication quality graphics and statistics. QIIME has been applied to studies based on billions of sequences from tens of thousands of samples. |
| [QUAST 4.0](https://wiki.cyverse.org/wiki/display/TUT/QUAST+4.0+Using+Atmosphere){target=_blank} | May 19, 2016 |	QUAST is a tool for evaluating genome assemblies by computing various metrics. |
| [rnaQUAST 1.1.0](https://wiki.cyverse.org/wiki/display/TUT/rnaQUAST+1.1.0+using+Atmosphere){target=_blank} | May 11, 2016 |	rnaQUAST is a tool for evaluating RNA-Seq assemblies using reference genome and gene data database. In addition, rnaQUAST is also capable of estimating gene database coverage by raw reads and de novo quality assessment using third-party software (STAR, TopHat, GMAP etc.). |
| [rnaQUAST 1.2.0](https://wiki.cyverse.org/wiki/display/TUT/rnaQUAST+1.1.0+using+Atmosphere){target=_blank} | May 19, 2016 |	rnaQUAST is a tool for evaluating RNA-Seq assemblies using reference genome and gene data database. In addition, rnaQUAST is also capable of estimating gene database coverage by raw reads and de novo quality assessment using third-party software (STAR, TopHat, GMAP etc.). |
| [Validate Workflow v0.9](https://wiki.cyverse.org/wiki/pages/viewpage.action?pageId=21139103){target=_blank} | Apr 19, 2016|	Learn to navigate the Validate Workflow. |
